lib/nokogiri/xml/document.rb

```diff
@@ -180,6 +184,38 @@ def initialize(*args) # :nodoc: # rubocop:disable Lint/MissingSuper
         @namespace_inheritance = false
       end
 
+      #
+      # :call-seq:
+      #   dup → Nokogiri::XML::Document
+      #   dup(level) → Nokogiri::XML::Document
+      #
+      # Duplicate this node.
+      #
+      # [Parameters]
+      # - +level+ (optional Integer). 0 is a shallow copy, 1 (the default) is a deep copy.
+      # [Returns] The new Nokogiri::XML::Document
+      #
+      def dup(level = 1)
+        copy = OBJECT_DUP_METHOD.bind_call(self)
```

Comment:
I don't particularly like the `bind_call` hack here and in and `#clone`, but I can't come up with anything better.